packages/programs/src/resolve-transaction-error.ts

```diff
@@ -0,0 +1,52 @@
+import type { Address } from '@solana/addresses';
+import type { Transaction } from '@solana/transactions';
+
+import { Program, ProgramWithErrors } from './program';
+
+/**
+ * Resolves a custom program error from a transaction error
+ * with logs using the provided list of registered programs.
+ * The original error is returned if the error cannot be
+ * resolved from the given programs.
+ *
+ * @param error The raw error to resolve containing the program logs.
+ * @param transaction The transaction that caused the error.
+ * @param programs The list of registered programs.
```

Comment:
What does it mean for a program to be registered? The term is used a few times but it looks like you just need an instance of `Program`

Reply:
That's a good point, I should probably change the wording around that.

Again, that's because, with Umi, you would _register_ your programs in a program repository by doing `umi.programs.add(myProgram)`.

Here though, we do not have control over how the end user keeps track of its programs which is why I accept an array of programs instead.

What do you think about the following?

```suggestion
 * @param programs The list of programs to go through when resolving the transaction error. They should ideally contain all programs the transaction is sending instructions to.
```

Reply:
Ah I see, the suggested change looks good! :)